### PR TITLE
Fixed Issue with Cordova v6 and +

### PR DESCRIPTION
**Pull request by luisvsilva**
Cordova now inserts icons as mipmap instead of drawables, so added routine to check for both resource types. Also removed package name parameter as this can be extracted from the current activity

### CODE REVIEW

**Comment by celiovasconcelos** [comment]
That's nice!

**Comment by luisvsilva** [comment]
If you are responsible / know the people responsible for the main pushlink site, sample code must change slightly:

https://www.pushlink.com/docs.xhtml#cordova

from:

`PushLink.start('com.myapp', 'my pushlink api key', 'device id', function() {`

to:

`PushLink.start('my pushlink api key', 'device id', function() {`

Also the sample application in:

https://github.com/pushlink/pushlink-phonegap-sample

Thanks!!!

**Comment by luisvsilva** [comment]
Also if you can bump the npm version and publish it it would be awesome!!!! Thanks

**Comment by celiovasconcelos** [comment]
Yes, we are preparing the `5.5.0.1` release in the next hours...

**Comment by luisvsilva** [comment]
I just noticed my last pull broke the build, I forgot to fix the unit tests. That has been done on the following PR:
[https://github.com/pushlink/pushlink-phonegap/pull/12](https://github.com/pushlink/pushlink-phonegap/pull/12)

Sorry

File: src/android/actions/StartAction.java [hunk 1]
```diff
@@ -11,15 +11,27 @@
...
+
+
   @Override
   public void execute(final CordovaInterface cordova, JSONObject arg, final CallbackContext callbackContext) throws Exception {
-    final String packageName = arg.getString(PACKAGE_NAME);
+    
     final String apiKey = arg.getString(API_KEY);
     final String deviceId = arg.getString(DEVICE_ID);
-    final int appIconId = cordova.getActivity().getResources().getIdentifier(ICON, "drawable", packageName);
+    final String packageName = cordova.getActivity().getApplicationContext().getPackageName();
```
Comment (rscarvalho) [inline src/android/actions/StartAction.java:26]: Is this a reliable way to get the application package name as it stated in the `AndroidManifest.xml`?

File: www/pushlink.js [hunk 2]
```diff
@@ -44,17 +44,15 @@ var PushLink = function() {
...
 /**
  * Starts PushLink connection
- * @param {string} packageName - The package name for the app
  * @param {string} apiKey - Your PushLink API key
  * @param {string} deviceId - The device id
  * @param {function} successCallback - A function to be called if the command succeeded
  * @param {function} errorCallback - A function to be called if the command failed
  */
-PushLink.prototype.start = function(packageName, apiKey, deviceId, successCallback, errorCallback) {
+PushLink.prototype.start = function(apiKey, deviceId, successCallback, errorCallback) {
```
Comment (rscarvalho) [inline www/pushlink.js:52]: I don't think changing the signature of the public api would be a good idea. We could play with `arguments` or checking the type of `successCallback` (or `deviceId`) to detect whether `packageName` whould be ignored.

**Review by rscarvalho — COMMENTED** [review]
Left two small comments there. These changes should be reversed to avoid breaking apps on upgrade of the library.

File: src/android/actions/StartAction.java (hunk 1 above)
Comment (celiovasconcelos) [reply src/android/actions/StartAction.java:26]: Yes it's for sure.

File: www/pushlink.js (hunk 2 above)
Comment (celiovasconcelos) [reply www/pushlink.js:52]: I think we need to change this. On the other hand we need to provide a good doc about the change.

**Comment by luisvsilva** [comment]
Rudolfo raises some valid points, if you want me to make that parameter
optional just let me know and I can prepare another PR.

If you guys decide to do a breaking change don't forget to bump the major
version in npm.

Thanks

Luis

**Comment by celiovasconcelos** [comment]
@rscarvalho 
Do you think the pushlink-phonegap should have its own version numbers?
For now, we have the same version name of the pushlink-android and it is causing problems naming/bumping the packages.
How about to create a compatbility matrix in the documentation?

Proposal:

| Jar File | Cordova| Comments|
| ------------- |:-------------:|-------------:|
| pushlink-android-5.5.2 | pushlink-phonegap-6.1.0 | `Not released yet` Fixed for Android 7 |
| pushlink-android-5.5.0 | pushlink-phonegap-6.0.0 | Fixed for Android 6 |
| pushlink-android-5.5.0 | pushlink-phonegap-5.5.0 |  |
| pushlink-android-5.4.4 | pushlink-phonegap-5.4.4 |  |

**Comment by celiovasconcelos** [comment]
@rscarvalho 
I'm noobie in `npm`, but i thing we should remove the `./bin/bump-version` in the script/version section of package.json. When I run `npm publish` it's increasing the name in the registry. I want the `npm publish` respect the `version` in package.json. 

UPDATE:
Ignore this message for while... maybe I did something wrongly...

**Comment by luisvsilva** [comment]
Hi Guys, any decision on this? Any chance to do a npm publish soon? I am
using a cloud building system so I am relying on npm for the build.

Again, if you want me to do another PR with the first parameter optional
let me know.

Thanks

Luis

**Comment by celiovasconcelos** [comment]
Hi,
I thought you were already using your local build.
Today we will release the `6.0.0` in `npm`.
Thank you!

**Pull request merged**